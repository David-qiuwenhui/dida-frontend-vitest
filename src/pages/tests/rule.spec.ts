import { it, expect, describe } from "vitest";
import {
  createPasswordLengthRule,
  createPasswordNotSameRule,
  createUsernameRule,
} from "../rules";

describe("validator rules", () => {
  it("should be return username rule", () => {
    expect(createUsernameRule()).matchSnapshot();
  });

  it("should be return password length rule", () => {
    expect(createPasswordLengthRule()).matchSnapshot();
  });

  it("should be return password length rule", () => {
    expect(createPasswordNotSameRule()).toMatchInlineSnapshot(`
      [
        {
          "message": "请再次输入密码",
          "required": true,
          "trigger": [
            "blur",
          ],
        },
        {
          "message": "两次输入的密码不一致",
          "trigger": [
            "blur",
          ],
          "validator": [Function],
        },
      ]
    `);
  });
});
