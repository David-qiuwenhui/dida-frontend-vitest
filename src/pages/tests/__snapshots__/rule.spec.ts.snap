// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`validator rules > should be return password length rule 1`] = `
[
  {
    "message": "请输入密码",
    "required": true,
    "trigger": [
      "blur",
    ],
  },
  {
    "message": "密码长度必须在 6 到 30 个字符之间",
    "trigger": [
      "blur",
    ],
    "validator": [Function],
  },
]
`;

exports[`validator rules > should be return username rule 1`] = `
[
  {
    "message": "请输入用户名",
    "required": true,
    "trigger": [
      "blur",
    ],
  },
  {
    "message": "用户名长度必须在 4 到 16 个字符之间",
    "trigger": [
      "blur",
    ],
    "validator": [Function],
  },
]
`;
